darts/explainability/shap_explainer.py

```diff
@@ -665,16 +666,28 @@ def _create_regression_model_shap_X(
         lags_past_covariates_list = self.model.lags.get("past")
         lags_future_covariates_list = self.model.lags.get("future")
 
-        X, _, indexes = _create_lagged_data(
-            target_series,
-            self.n,
-            past_covariates,
-            future_covariates,
-            lags_list,
-            lags_past_covariates_list,
-            lags_future_covariates_list,
-            is_training=False,
-        )
+        # ensure list of TimeSeries format
+        if isinstance(target_series, TimeSeries):
+            target_series = [target_series]
+            past_covariates = [past_covariates] if past_covariates else None
+            future_covariates = [future_covariates] if future_covariates else None
+        X, indexes = [], []
+        for i, target in enumerate(target_series):
```

Comment:
So part of the motivation behind this was so that `RegressionModel` would have access to the unconcatenated '`X` blocks' corresponding to each tabularized `TimeSeries`. By having access to the unconcatenated blocks, `_add_static_covariates` doesn't need to 'back-compute' how many lagged features were contributed to `X` by each `TimeSeries`, thereby effectively decoupling the logic of `_add_static_covariates` and `create_lagged_data`. An alternative approach to achieve this same effect would be to allow `create_lagged_data` to accept `Sequence[TimeSeries]` inputs, but have it return `Sequence[TimeSeries]` for `X` and `y`, meaning that the only repeated piece of code would be calling `np.concatenate` after calling `create_lagged_features`. What are your thoughts on this?

Reply:
OK thanks for the explanation, makes sense.
One alternative, to remove the need for the caller to worry about concatenation, could be to introduce a new function `create_tabular_data(all_series: Sequence[TimeSeries]) -> np.ndarray`, which would be in charge of iterating over series and calling `create_lagged_data()` and `_add_static_covariates()` on each.
But it's not very important IMO and we can also leave it like that for the time being.